modules/account-tokenizer/src/lib.rs

```diff
@@ -0,0 +1,448 @@
+					"Account Tokenizer: Transferred funds from treasury to create class. result: {:?}",
+					res
+				);
+
+				// Use storage version to ensure we only register NFT class once.
+				let class_id = T::NFTInterface::next_class_id();
+				let res = T::NFTInterface::create_class(&class_id, &T::PalletAccount::get(), &T::PalletAccount::get());
+				log::debug!("Account Tokenizer: Created NFT class. result: {:?}", res);
+
+				let res = T::NFTInterface::set_class_properties(
+					&class_id,
+					Properties(
+						ClassProperty::Transferable
+							| ClassProperty::Burnable | ClassProperty::Mintable
+							| ClassProperty::ClassPropertiesMutable,
+					)
+					.into(),
+				);
+				log::debug!("Account Tokenizer: Set NFT class property. result: {:?}", res);
+
+				// Sets NFT class ID storage
+				NFTClassId::<T>::put(class_id);
+
+				// Upgrade storage versino so NFT class is only created once.
+				StorageVersion::new(1).put::<Self>();
+				<T as Config>::WeightInfo::initialize_nft_class()
+			} else {
+				0
+			}
+		}
+
+		// ensure that MintFee is >= NFT's mint fee.
+		fn integrity_test() {
+			sp_std::if_std! {
+				sp_io::TestExternalities::new_empty().execute_with(||
+					assert!(
+						T::MintFee::get() >= T::NFTInterface::base_mint_fee()
+					));
+			}
+		}
+	}
+
+	#[pallet::pallet]
+	#[pallet::without_storage_info]
+	pub struct Pallet<T>(_);
+
+	#[pallet::call]
+	impl<T: Config> Pallet<T> {
+		/// Request to mint an Account Token. Called after the user of the same Account Id has given
+		/// the proxy control of an account to the parachain account.
+		///
+		/// Params:
+		/// 	- `account`: The account ID of the anonymous proxy.
+		/// 	- `height`: The block number in which the anonymous proxy is generated.
+		/// 	- `ext_index`: The index, in the block, of the extrinsics that generated the anonymous
+		///    proxy.
+		/// 	- `index`: The index of the anonymous proxy.
+		#[pallet::weight(< T as Config >::WeightInfo::request_mint())]
+		#[transactional]
+		pub fn request_mint(
+			origin: OriginFor<T>,
+			account: T::AccountId,
+			height: T::BlockNumber,
+			ext_index: u32,
+			index: u16,
+		) -> DispatchResult {
+			let who = ensure_signed(origin)?;
+
+			// Checks if the account is an anonymous proxy of the caller.
+			// hard coded for `ProxyType::Any`. No other proxy type is allowed
+			let entropy =
+				(b"modlpy/proxy____", &who, height, ext_index, &PROXYTYPE_ANY, index).using_encoded(blake2_256);
+			let derived_account: T::AccountId = Decode::decode(&mut TrailingZeroInput::new(entropy.as_ref()))
+				.expect("infinite length input; no invalid inputs for type; qed");
+			// ensures signer also spawned anonymous proxy
+			ensure!(&account == &derived_account, Error::<T>::FailedAnonymousCheck);
+
+			// Charge the user fee and lock the deposit.
+			T::Currency::transfer(&who, &T::TreasuryAccount::get(), T::MintFee::get(), KeepAlive)?;
+			T::Currency::reserve_named(&RESERVE_ID, &who, T::MintRequestDeposit::get())?;
```

Comment:
I don't think we need deposit here if `ForeignStateQuery` already takes one

Reply:
After discussion, we decided to keep the deposit in this module, due to its unique business logic